src/core/main.c

```diff
@@ -685,6 +685,29 @@ static int parse_config_file(void) {
         return 0;
 }
 
+static int reload_manager_defaults(Manager *m) {
+        int r;
+
+        assert(m);
+
+        r = parse_config_file();
+        if (r < 0)
+               return r;
+
+        m->default_std_output = arg_default_std_output;
+        m->default_std_error = arg_default_std_error;
+        m->default_timeout_start_usec = arg_default_timeout_start_usec;
+        m->default_timeout_stop_usec = arg_default_timeout_stop_usec;
+        m->default_restart_usec = arg_default_restart_usec;
+        m->default_start_limit_interval = arg_default_start_limit_interval;
+        m->default_start_limit_burst = arg_default_start_limit_burst;
+
+        manager_set_default_rlimits(m, arg_default_rlimit);
+        manager_environment_add(m, NULL, arg_default_environment);
+
+        return 0;
+}
+
```

Comment:
Could you rename this function to `manager_set_default()` or something and then also call it from `main()` to get rid of the code duplication?